src/Microsoft.Extensions.Caching.Memory/MemoryCache.cs

```diff
@@ -302,6 +258,68 @@ private void EntryExpired(CacheEntry entry)
             StartScanForExpiredItems();
         }
 
+        private void SetEntry(CacheEntry entry)
+        {
+            var utcNow = _clock.UtcNow;
+
+            DateTimeOffset? absoluteExpiration = null;
+            if (entry._absoluteExpirationRelativeToNow.HasValue)
+            {
+                absoluteExpiration = utcNow + entry._absoluteExpirationRelativeToNow;
+            }
+            else if (entry._absoluteExpiration.HasValue)
+            {
+                if (entry._absoluteExpiration <= utcNow)
+                {
+                    throw new ArgumentOutOfRangeException(
```

Comment:
SetEntry is only called from Dispose which should never throw. If you remove this I think CheckExpired will handle it as already expired.